docker/Dockerfile.ng

```diff
@@ -0,0 +1,213 @@
+
+RUN set -ex \
+    \
+    # Add build dependencies
+    && apk add --no-cache --virtual .build-dependencies \
+        build-base \
+        linux-headers \
+        postgresql-dev \
+        mariadb-dev \
+        sqlite-dev \
+    \
+    # Add runtime dependencies
+    && apk add --no-cache \
+        tzdata \
+        libpq \
+        mariadb-client-libs \
+        libcurl \
+        su-exec \
+        git \
+        #sudo \
+    \
+    # Hack to avoid "fatal: Not a git repository" from bundler
+    && git init \
+    \
+    # Install Gemfile dependencies
+    #&& chown -R huginn:huginn "$HUGINN_HOME" \
+    #&& su-exec huginn echo "gem 'sqlite3', '~> $SQLITE_VERSION'" >> "$HUGINN_HOME/Gemfile" \
+    && echo "gem 'sqlite3', '~> $SQLITE_VERSION'" >> "$HUGINN_HOME/Gemfile" \
+    && INSTALL_ALL_DBS=true bundle install --without test development --jobs 4 \
+    #--deployment # See http://bundler.io/bundle_install.html
+    #--path vendor/bundle --jobs 4
+    \
+    # Cleanup ruby caches
+    && INSTALL_ALL_DBS=true bundle clean \
+    #&& rm -rf /root/.gem/specs/* \
+    #&& rm -rf /root/.bundle/cache/* \
+    #&& rm -rf /usr/local/lib/ruby/gems/2.4.0/cache/* \
+    \
+    # Cleanup transient dependencies/caches
+    && apk del .build-dependencies \
+    && rm -rf /var/cache/apk/* \
+    && rm -rf /var/lib/apk/* \
+    && rm -rf /etc/apk/cache/*
+
+# TODO: git init is a hack to avoid "fatal: Not a git repository (or any of the parent directories): .git" when runnings rails/etc
+#   Better to just figure how to solve it properly..
+
+# TODO: Whats the benefit of running this to the /vendor dir? Can we chagne it? Will it use system rubyracer either way?
+# TODO: Should we be bundle installing with --deployment / --path vendor? Or to system?
+# TODO: Is there any way/benefit to specify the DB we want in an efficient way, so we don't need the other overhead files? Does it even make much size difference?
+# TODO: Do we need to keep git as a runtime dependency? It adds a solid chunk to our image size..
+
+# TODO: Add postgresql-dev, then delete all files but these? https://packages.debian.org/sid/i386/libpq-dev/filelist
+# TODO: Add mariadb-dev, then delete all files but these? http://packages.ubuntu.com/trusty/i386/libmysqlclient-dev/filelist
+# http://stackoverflow.com/questions/16304311/use-mariadb-instead-of-mysql-in-my-rails-project
+# https://packages.debian.org/sid/i386/libmariadbd-dev/filelist
+
+# Old Dependencies for reference
+#apt-get update
+#$minimal_apt_get_install build-essential checkinstall git-core \
+#  zlib1g-dev libyaml-dev libssl-dev libgdbm-dev libreadline-dev \
+#  libncurses5-dev libffi-dev libxml2-dev libxslt-dev libcurl4-openssl-dev libicu-dev \
+#  graphviz libgraphviz-dev \
+#  libmysqlclient-dev libpq-dev libsqlite3-dev \
+#  ruby2.3 ruby2.3-dev
+
+# TODO: Put together a little script that will easily parse out all the .env.example lines to copy into here
```

Comment:
I like the idea of dropping the `.env` file in the docker image. We need to decide how we handle backwards compatibility, the current solution with .env` sometimes requires double escaped strings which will probably not work without converting the values here.